drivers/flash/flash_stm32h7x.c

```diff
@@ -767,7 +767,7 @@ static int stm32h7_flash_init(const struct device *dev)
 	}
 #endif
 
-	return flash_stm32h7_write_protection(dev, false);
+	return flash_stm32h7_write_protection(dev, true);
```

Comment:
I did the same change in `flash_stm32` driver 2 years ago #52371

I think, the initial state is that control register is locked, so we can just return 0 here. Could you check that in STM32H7 reference manual?

Reply:
Makes sense. The reset state is locked, so I agree we can just `return 0;` here.